source/io/module_eggShell.js

```diff
@@ -250,6 +250,14 @@ var assignEggShell;
             return dispatchFunction(typeVisitor, valueRef, data);
         };
 
+        Module.eggShell.checkAndResetValueRefNeedsUpdateFlag = publicAPI.eggShell.checkAndResetValueRefNeedsUpdateFlag = function (valueRef) {
```

Comment:
I think we can simplify the name a bit. Maybe just `needsUpdate`.
in the context of usage:
```
var valueRef = vireo.eggShell.findValueRef('MyVI', 'dataItem_Boolean');
if(vireo.eggShell.needsUpdate(valueRef)) {
   ...
}
```

And that can be the only function on the public api. We can avoid assigning `checkValueRefNeedsUpdateFlag_SkipReset_` to the  `publicAPI.eggShell` object to make sure it's not used except for debugging. We could probably also simplify the name, maybe `needsUpdateSkipReset`?

We will still assign the debug function to the `Module.eggShell` object and the name helps makes sure it won't get used except for debugging and [maybe some tests](https://github.com/ni/VireoSDK/blob/0e94fa522f9b0564bf9713eb8386a229ec8c7ce1/test-it/karma/publicapi/ExecuteSlicesUntilClumpsFinshed.Test.js#L97) :)

Reply:
I tried to leave checkValueRefNeedsUpdateFlag_SkipReset_ out of publicAPI.eggShell, but then I wasn't sure how to call it from the karma test. Can you provide guidance on that?

Reply:
Ah, never mind.  I see the 'maybe some tests' link you just gave already provides an example.

Reply:
I'm not a fan of long names either but I don't like what looks like a simple predicate function actually having a side effect of resetting the value without any indication that it does so.
How about something like:
`if(vireo.eggShell.needsUpdate_testAndReset(valueRef)) {
}
`

Reply:
Yea I agree about not hiding side-effects.

During the api refactor we tried to cleanup the api to be consistently camelCase and not mix formats.

What about `needsUpdateAndReset` or `needsUpdateAndResetFlag` or `needsUpdateAndResetDataItem`?

Reply:
TestAndSet, TestAndReset denote the atomic operation, so removing Test from the name will lose that clarity.

Reply:
My suggestion was before the latest updates to testAndResetNeedsUpdateFlag and testNeedsUpdateFlagWithoutReset. The newest updates saying they test and reset are good to me as well.